### PR TITLE
TEST: Refactor the ucp_client_server example to use the new API.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10104/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10124/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10125/ for details (Mellanox internal link).

File: test/examples/ucp_client_server.c [hunk 1]
```diff
@@ -42,12 +42,12 @@ static uint16_t server_port = 13337;
 
 
 /**
- * Server context to be used in the user's accept callback.
- * It holds the server's endpoint which will be created upon accepting a
- * connection request from the client.
+ * Server's application context to be used in the user's connection request
+ * callback.
+ * It holds the server's ucp context which is created at the init stage.
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: minor: "at the" -> "during the"

File: test/examples/ucp_client_server.c [hunk 2]
```diff
@@ -363,19 +325,129 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
...
+{
+    ucx_server_ctx_t *context = arg;
+    ucp_ep_h         ep;
+    ucp_ep_params_t  ep_params;
+    ucs_status_t     status;
+    ucp_worker_h     new_worker;
+    int ret;
+
+    /* Create a new worker */
+    ret = init_worker(context->ctx, &new_worker);
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: i would not create a worker here. during init phase, create 2 workers: "connection_worker" and "data_worker", and use "data_worker" here to create an endpoint.
it's not a good practice to create worker for each new connection, so we also don't want to set such example.

File: test/examples/ucp_client_server.c [hunk 3]
```diff
@@ -363,19 +325,129 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
...
+
+    /* Client-Server communication via Stream API */
+    send_recv_stream(new_worker, ep, 1);
+
+    /* Close the endpoint to the client and destroy the new worker since the
+     * communication to the client ended */
+    ep_close(new_worker, ep);
+    ucp_worker_destroy(new_worker);
+
+    printf("Waiting for connection...\n");
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: "Waiting for another connection..."

File: test/examples/ucp_client_server.c [hunk 4]
```diff
@@ -363,19 +325,129 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
...
+}
+
+/**
+ * Initialize the server side. The server starts listening on the set address.
+ */
+static int start_server(ucp_worker_h ucp_worker, ucx_server_ctx_t *context,
+                        ucp_listener_h *listener)
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: listener->listener_p

File: src/ucp/core/ucp_worker.c [hunk 5]
```diff
@@ -501,7 +501,8 @@ ucs_status_t ucp_worker_set_ep_failed(ucp_worker_h worker, ucp_ep_h ucp_ep,
             rsc_index = ucp_ep_get_rsc_index(ucp_ep, lane);
             tl_rsc    = &worker->context->tl_rscs[rsc_index].tl_rsc;
             ucs_error("error '%s' will not be handled for ep %p - "
-                      UCT_TL_RESOURCE_DESC_FMT, ucs_status_string(status), ucp_ep,
+                      UCT_TL_RESOURCE_DESC_FMT " since no error callback is installed",
```
Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: should we add "since no error callback is installed" to line 509 as well?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10186/ for details (Mellanox internal link).

**Pull request merged**